bitcoin/contrib/test.sh

```diff
@@ -81,7 +81,10 @@ cargo run --example taproot-psbt --features=bitcoinconsensus
 
 # Build the docs if told to (this only works with the nightly toolchain)
 if [ "$DO_DOCS" = true ]; then
-    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly rustdoc --all-features -- -D rustdoc::broken-intra-doc-links -D warnings || exit 1
+    # We use rustdoc so that we can check for broken links.
+    RUSTDOCFLAGS="--cfg docsrs" cargo +nightly rustdoc --all-features -- -D rustdoc::broken-intra-doc-links -D warnings
+    # This, in unison with the command above, checks that we feature guarded docs imports correctly.
+    cargo +nightly doc
```

Comment:
If we're doing this we should still have all features and deny warnings, just `docsrs` disabled. And probably test it on stable instead.

Reply:
Yes and face palm for `--all-features` and deny warnings. For test on stable, seems reasonably, will require more thinking though since `DO_DOCS` is only run on nightly, perhaps we need to agree on what is expected from each toolchain since this is similar to the discussion on features only working on some toolchains (https://github.com/rust-bitcoin/rust-bitcoin/issues/1498#issuecomment-1364409850)

Reply:
I think just adding `+stable` should work? It's a bit ugly hack, yes. Or just split it up into `DO_DOCSRS` and `DO_DOCS`.

Reply:
Unless I'm missing something the `stable` toolchain won't be installed on the VM when we use `DO_DOCS`. I used your `DO_DOCSRS` idea, adding a `DO_DOCS` for stable builds.